### PR TITLE
Tests: Fix t.Parallel() errors in util package

### PR DESCRIPTION
**Pull request by michaeldiamant**
Enables https://github.com/kunwardeep/paralleltest on `util` by fixing linter warnings.  Incrementally moves the ball towards greater unit test parallelization, which reduces local + CI test durations.

I vetted for flakiness by running:
```
go test -race -count 10 ./util/...
```

Notes:
* The PR does _not_ provide a meaningful reduction in package test duration because sequential tests `github.com/algorand/go-algorand/util/db` in dominate test time.  
* The PR does _not_ intend to rework sequential tests into parallel tests.  My objective is to enable already available parallelism throughout go-algorand.  And then make a pass through remaining sequential tests to see if/where more parallelism exists.

### CODE REVIEW

**Review by jdtzmn — APPROVED** [review]
Changes look good! I have a hunch that some of the db tests can be parallelized by using different db names so I pushed some commits to see whether tests pass in CI.

**Review by algochoi — APPROVED** [review]
This looks good on my local tests + checked that the linter catches all instances in the `util` package.

File: util/metrics/metrics_test.go [hunk 1]
```diff
@@ -110,14 +111,17 @@ func TestSanitizeTelemetryName(t *testing.T) {
 		{in: "-my-metric-name", out: "_my-metric-name"},
 		{in: `label-counter:label="a label value"`, out: "label-counter_label__a_label_value_"},
 	} {
+		tc := tc
 		t.Run(tc.in, func(t *testing.T) {
+			t.Parallel()
```
[the remark this answers is not in the document]
Comment (algochoi) [reply util/metrics/metrics_test.go:116]: It looks like the linter will call out non-parallel tests in the range: https://github.com/kunwardeep/paralleltest#missing-tparallel-in-the-range-method

**Review by algochoi — COMMENTED** [review]
@cce 
I addressed the PR comments by pushing to this branch. I removed parallelization from tests that may have timing dependencies or are "non-short" tests. I also agree that tests that call benchmarks are not a good candidate for parallelizing right now.

**Comment by bbroder-algo** [comment]
Is this stripped down to only tests that will not have multiparallelism issues now? @algochoi

**Comment by algochoi** [comment]
> Is this stripped down to only tests that will not have multiparallelism issues now? @algochoi

Yes, I addressed the parallelism/timing dependency concerns in PR feedback

**Review by bbroder-algo — APPROVED** [review]
@cce

**Pull request closed without merging**